public/app/plugins/panel/geomap/components/MeasureOverlay.tsx

```diff
@@ -0,0 +1,264 @@
+import { css } from '@emotion/css';
+import Map from 'ol/Map';
+import { Geometry, LineString, Point } from 'ol/geom';
+import { Draw, Modify } from 'ol/interaction';
+import { Vector as VectorLayer } from 'ol/layer';
+import { Vector as VectorSource } from 'ol/source';
+import { getArea, getLength } from 'ol/sphere';
+import React, { useMemo, useState } from 'react';
+import tinycolor from 'tinycolor2';
+
+import { formattedValueToString, GrafanaTheme, SelectableValue } from '@grafana/data';
+import { IconButton, RadioButtonGroup, Select, stylesFactory } from '@grafana/ui';
+import { config } from 'app/core/config';
+
+import { style, labelStyle, tipStyle, modifyStyle, segmentStyle } from '../globalStyles';
+import { MapMeasure, MapMeasureOptions, measures } from '../utils/measure';
+
+type Props = {
+  map: Map;
+  menuActiveState: (value: boolean) => void;
+};
+
+export const MeasureOverlay = ({ map, menuActiveState }: Props) => {
+  const measureStyle = getStyles(config.theme);
+
+  // Menu State Management
+  const [firstLoad, setFirstLoad] = useState<boolean>(true);
+  const [menuActive, setMenuActive] = useState<boolean>(false);
+
+  // Options State
+  const [options, setOptions] = useState<MapMeasureOptions>({
+    action: measures[1].value!,
+    unit: measures[1].units[0].value!,
+    unitLabel: measures[1].units[0].labelOverlay,
+  });
+  const unit = useMemo(() => {
+    const action = measures.find((m: MapMeasure) => m.value === options.action) ?? measures[0];
+    const current = action.getUnit(options.unit);
+    const fn = action.value === 'area' ? getArea : getLength;
+    const measure = (geo: Geometry) => {
+      const v = fn(geo);
+      return formattedValueToString(current.format(v));
+    };
+    return {
+      current,
+      options: action.units,
+      measure,
+    };
+  }, [options]);
+
+  const clearPrevious = true;
+  const showSegments = false;
+
+  function toggleMenu() {
+    setMenuActive(!menuActive);
+    // Lift menu state
+    // TODO: consolidate into one state
+    menuActiveState(!menuActive);
+    if (menuActive) {
+      map.removeInteraction(draw);
+      vector.set('visible', false);
+    } else {
+      if (firstLoad) {
+        // Initialize on first load
+        setFirstLoad(false);
+        map.addLayer(vector);
+        map.addInteraction(modify);
+      }
+      vector.set('visible', true);
+      map.removeInteraction(draw); // Remove last interaction
+      const a = measures.find((v: MapMeasure) => v.value === options.action) ?? measures[0];
+      addInteraction(map, a.geometry, showSegments, clearPrevious);
+    }
+  }
+
+  return (
+    <div className={`${measureStyle.infoWrap} ol-unselectable ol-control`} style={{ backgroundColor: '#22252b' }}>
```

Comment:
be sure to check how it looks in both light+dark theme.  Is the explicit style here because of the interaction with `ol-control`?

Reply:
Yep, I believe this is to override the default highlighting behavior, but I will double check.